qiskit/mapper/_layout.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+A two-ways dict that represent a layout.
+
+Layout is the relation between logical (qu)bits and physical wires.
```

Comment:
Throughout the PR, can you please change "logical" to "virtual"? In effect those are virtual qubits that don't yet have a binding to physical qubits.

In the literature "logical" qubits are used in the context of error correction, so we shouldn't conflate the terms.  Thanks @awcross1 for reminding me.

Reply:
Yeap... I also did not like the "logical". In df55c03, I removed `logical` fully (and `physical` in 049f5aa). Layout is a map between (qu)bits and wires. I think is more clear.